resources/providers/azurelib/inventory/postgresql_provider.go

```diff
@@ -144,3 +164,73 @@ func (p *psqlProvider) ListFlexiblePostgresConfigurations(ctx context.Context, s
 
 	return assets, nil
 }
+
+func (p *psqlProvider) ListSinglePostgresFirewallRules(ctx context.Context, subID, resourceGroup, serverName string) ([]AzureAsset, error) {
+	paged, err := p.client.AssetSingleServerFirewallRules(ctx, subID, resourceGroup, serverName, nil, nil)
+	if err != nil {
+		return nil, err
+	}
+
+	configs := lo.FlatMap(paged, func(p armpostgresql.FirewallRulesClientListByServerResponse, _ int) []*armpostgresql.FirewallRule {
+		return p.Value
+	})
+
+	assets := make([]AzureAsset, 0, len(configs))
+	for _, fr := range configs {
+		if fr == nil || fr.Properties == nil {
+			continue
+		}
+
+		assets = append(assets, AzureAsset{
+			Id:       pointers.Deref(fr.ID),
+			Name:     pointers.Deref(fr.Name),
+			Location: assetLocationGlobal,
+			Properties: map[string]any{
+				"name":           pointers.Deref(fr.Name),
+				"startIPAddress": pointers.Deref(fr.Properties.StartIPAddress),
```

Comment:
My only concern is that we have a generic firewall rules provider at this level, but we filter only the data with the end goal in mind (the enricher and, thus, the rules this will be used for).

Of course, we could always refactor later, but would it be better if the provider provided the whole asset? 

(not necessary to change just a brainstorming point)

Reply:
What data is being filtered?

Reply:
Yes, sorry, it wasn't clear; by data, I mean the complete information/fields included inside each Azure struct, for example, `FirewallRule`.

It does not transform, for example, all `.Properties` into `AzureAsset` but instead only the ones it will use later in the enricher and rule side.  

So we have a provider that is called Postgre SQL Provider and, for example, a function that is called `ListSinglePostgresFirewallRules`, but actually, it does not provide the complete transformed object of `armpostgresql.FirewallRule` but only a subset of the properties/fields included in it. 

As I said, it's not necessarily something to change here, but perhaps we can reevaluate later - regarding providers and if they should provide the complete set of fields or a sub-set.

Reply:
Sorry, apparently, I mixed up the structs 😞  (flexible server `ConfigurationProperties` with the code introduced in this PR) and thought that a subset of fields was used. 

False alarm. Let's discuss, away from this PR, the possibility of adding possible missing fields in other structs as well.